build-logic/integration-testing/src/main/kotlin/gradlebuild/integrationtests/shared-configuration.kt

```diff
@@ -205,26 +207,28 @@ fun IntegrationTest.setUpAgentIfNeeded(testType: TestType, executer: String) {
 private
 fun IntegrationTest.addDebugProperties() {
     // TODO Move magic property out
-    if (project.hasProperty("org.gradle.integtest.debug")) {
+    val environmentExtension = project.getBuildEnvironmentExtension()
+    if (environmentExtension.integtestDebug.isPresent) {
         systemProperties["org.gradle.integtest.debug"] = "true"
         testLogging.showStandardStreams = true
     }
     // TODO Move magic property out
-    if (project.hasProperty("org.gradle.integtest.verbose")) {
+    if (environmentExtension.integtestVerbose.isPresent) {
         testLogging.showStandardStreams = true
     }
     // TODO Move magic property out
-    if (project.hasProperty("org.gradle.integtest.launcher.debug")) {
+    if (environmentExtension.integtestLauncherDebug.isPresent) {
```

Comment:
❌  We should move/remove the TODO if the corresponding string literal starts to live in a new place

Reply:
Kept, since I removed `integtestLauncherDebug` and I read the magic property via `providers.gradleProperty`